Unregister does not reset IsRegistered to false

If I call `RegisterInstance(...)` and then `Unregister()`, `IsRegistered` continues to return true.

Any code using this to perform conditional registration checks will subsequently not register, and then crash when attempting to resolve an MSBuild assembly.

Hi everyone,
I found this library completely useful. But I'm having a hard problem to solve.

Let's consider I have a method "ExecuteCommand" for executing commands (using the classical ProcessStartInfo and Process instances).
Then, I create a project programmatically using dotnet commands, after that, I want to open that solution, and after that, I want to run again the solution.
This is the scenario in my source code:

`
ExecuteCommand("dotnet new sln -n MySolution");
ExecuteCommand("dotnet new console -n MyProject -f netcoreapp3.1");
ExecuteCommand("dotnet sln MySolution.sln add MyProject\\MyProject.csproj");
ExecuteCommand("dotnet run"); // In this case everything goes fine!
// Here I want to analyze the solution (or doing some code analysis stuff)
MSBuildLocator.RegisterDefaults();
var msBuildWorkspace = MSBuildWorkspace.Create();
var mySolution = msBuildWorkspace.OpenSolutionAsync(Configuration.solutionFiles.solutionPath).Result;
// Here I have my solution, and that's okay, but I want to run again the same solution
ExecuteCommand("dotnet run"); // In this case when I execute I have the following exception...
`

C:\Program Files\dotnet\sdk\3.1.301\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018: Unexpected error in the task "GenerateDepsFile". [C:\Users\myuser\AppData\Local\Temp\TestProject\TestProject\TestProject.csproj]
C:\Program Files\dotnet\sdk\3.1.301\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.Sdk.targets(194,5): error MSB4018: System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.DotNet.PlatformAbstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system can not found the specified file. ...

After analyzing everything, the problem starts with "MSBuildLocator.RegisterDefaults();".
If I execute "dotnet run" immediately after this line I have the same error.
In this case, I wanted to unregister everything related to MSBuildLocator and I can't, even if I add MSBuildLocation.Unregister().

Anyone knows how can I solve this situation?
Thanks in advance :)

--- a/src/MSBuildLocator/MSBuildLocator.cs
+++ b/src/MSBuildLocator/MSBuildLocator.cs
@@ -290,19 +290,6 @@ namespace Microsoft.Build.Locator
         /// </remarks>
         public static void Unregister()
         {
-            if (!IsRegistered)
-            {
-                var error = $"{typeof(MSBuildLocator)}.{nameof(Unregister)} was called, but no MSBuild instance is registered." + Environment.NewLine +
-                            $"Ensure that {nameof(RegisterInstance)}, {nameof(RegisterMSBuildPath)}, or {nameof(RegisterDefaults)} is called before calling this method." + Environment.NewLine +
-                            $"{nameof(IsRegistered)} should be used to determine whether calling {nameof(Unregister)} is a valid operation.";
-                throw new InvalidOperationException(error);
-            }
-
-#if NET46
-            AppDomain.CurrentDomain.AssemblyResolve -= s_registeredHandler;
-#else
-            AssemblyLoadContext.Default.Resolving -= s_registeredHandler;
-#endif
         }
 
         /// <summary>

--- a/src/MSBuildLocator/MSBuildLocator.cs
+++ b/src/MSBuildLocator/MSBuildLocator.cs
@@ -283,11 +283,8 @@ namespace Microsoft.Build.Locator
         }
 
         /// <summary>
-        ///     Remove assembly resolution previously registered via <see cref="RegisterInstance" />, <see cref="RegisterMSBuildPath" />, or <see cref="RegisterDefaults" />.
+        ///     This has no effect and exists only for backwards compatibility. Calling it is unnecessary.
         /// </summary>
-        /// <remarks>
-        ///     This will automatically be called once all supported assemblies are loaded into the current AppDomain and so generally is not necessary to call directly.
-        /// </remarks>
         public static void Unregister()
         {
         }

--- a/src/MSBuildLocator/MSBuildLocator.cs
+++ b/src/MSBuildLocator/MSBuildLocator.cs
@@ -8,6 +8,7 @@ using System.IO;
 using System.Linq;
 using System.Reflection;
 using System.Text;
+using System.ComponentModel;
 
 #if NETCOREAPP
 using System.Runtime.Loader;
@@ -285,6 +286,7 @@ namespace Microsoft.Build.Locator
         /// <summary>
         ///     This has no effect and exists only for backwards compatibility. Calling it is unnecessary.
         /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
         public static void Unregister()
         {
         }

--- a/src/MSBuildLocator/MSBuildLocator.cs
+++ b/src/MSBuildLocator/MSBuildLocator.cs
@@ -3,12 +3,12 @@
 
 using System;
 using System.Collections.Generic;
+using System.ComponentModel;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Reflection;
 using System.Text;
-using System.ComponentModel;
 
 #if NETCOREAPP
 using System.Runtime.Loader;

